ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -14022,6 +14032,26 @@ void SemanticErrorChecker::reportError(int errNo, const char * format, ...)
     va_end(args);
 }
 
+void SemanticErrorChecker::reportWarning(WarnErrorCategory category, int warnNo, const char * format, ...)
+{
+    ECLlocation location;
+    ForEachItemInRev(i, locations)
+    {
+        if (location.extractLocationAttr(&locations.item(i)))
+            break;
+    }
+    va_list args;
+    va_start(args, format);
+
+    StringBuffer msg;
+    msg.valist_appendf(format, args);
+    ErrorSeverity severity = queryDefaultSeverity(category);
+    Owned<IError> err = createError(category, severity, warnNo, msg, str(location.sourcePath), location.lineno, location.column);
+    mapper.mapError(err);
```

Comment:
This looks wrong - this will be leaking the mapped error.  I would expect:

```
        Owned<IError> mappedError = mapper->mapError(err);
        mapper.report(mappedError);
```
E.g. some of the examples in ecl/regress no longer generate the errors/warnings they should.